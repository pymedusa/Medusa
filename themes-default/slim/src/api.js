import axios from 'axios';

// This should be more dynamic. As now when we change the apiKey in config/general. This won't work anymore.
// Because of this, a page reload is required.
export const webRoot = document.body.getAttribute('web-root');
export const apiKey = document.body.getAttribute('api-key');

/**
 * Api client based on the axios client, to communicate with medusa's web routes, which return json data.
 */
export const apiRoute = axios.create({
    baseURL: webRoot + '/',
    timeout: 60000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
    }
});

/**
 * Api client based on the axios client, to communicate with medusa's api v1.
 */
export const apiv1 = axios.create({
    baseURL: webRoot + '/api/v1/' + apiKey + '/',
    timeout: 30000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
    }
});

/**
 * Api client based on the axios client, to communicate with medusa's api v2.
 */
export const api = axios.create({
    baseURL: webRoot + '/api/v2/',
    timeout: 30000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        'X-Api-Key': apiKey
    }
});
