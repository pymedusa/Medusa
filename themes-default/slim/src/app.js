import Vue from 'vue';
import Vuex, { mapState } from 'vuex';
import VueRouter from 'vue-router';
import AsyncComputed from 'vue-async-computed';
import ToggleButton from 'vue-js-toggle-button';
import Snotify from 'vue-snotify';
import Truncate from 'vue-truncate-collapsed';
import store from './store';
import router from './router';
import { isDevelopment } from './utils';
import {
    AnidbReleaseGroupUi,
    AppHeader,
    AppLink,
    Asset,
    Backstretch,
    Config,
    DisplayShow,
    FileBrowser,
    LanguageSelect,
    NamePattern,
    PlotInfo,
    RootDirs,
    ScrollButtons,
    SelectList,
    ShowSelector
} from './components';

Vue.config.devtools = true;
Vue.config.performance = true;

Vue.use(Vuex);
Vue.use(VueRouter);
Vue.use(AsyncComputed);
Vue.use(ToggleButton);
Vue.use(Snotify);

// Load x-template components
window.components.forEach(component => {
    if (isDevelopment) {
        console.debug(`Registering ${component.name}`);
    }
    Vue.component(component.name, component);
});

// Global components
const globalComponents = [
    AnidbReleaseGroupUi,
    AppHeader,
    AppLink,
    Asset,
    Backstretch,
    Config,
    DisplayShow,
    FileBrowser,
    LanguageSelect,
    NamePattern,
    PlotInfo,
    RootDirs,
    ScrollButtons,
    SelectList,
    ShowSelector
];

globalComponents.forEach(component => {
    Vue.component(component.name, component);
});

const app = new Vue({
    name: 'App',
    store,
    router,
    components: {
        Truncate
    },
    data() {
        return {
            globalLoading: false,
            pageComponent: false
        };
    },
    computed: Object.assign(mapState(['auth', 'config']), {}),
    mounted() {
        if (isDevelopment) {
            console.log('App Mounted!');
        }

        if (!document.location.pathname.includes('/login')) {
            const { $store } = this;
            $store.dispatch('login', { username: window.username });
            $store.dispatch('getConfig');

            if (isDevelopment) {
                console.log('App Loaded!');
            }
        }
    }
}).$mount('#vue-wrap');

export default app;
