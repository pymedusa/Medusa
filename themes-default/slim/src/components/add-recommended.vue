<template>
    <div id="addShowPortal">
        <app-link href="addShows/trendingShows/?traktList=anticipated" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addtrakt"></div></div>
            <div class="buttontext">
                <h3>Add From Trakt Lists</h3>
                <p>For shows that you haven't downloaded yet, this option lets you choose from a show from one of the Trakt lists to add to Medusa .</p>
            </div>
        </app-link>

        <app-link href="addShows/popularShows/" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addimdb"></div></div>
            <div class="buttontext">
                <h3>Add From IMDB's Popular Shows</h3>
                <p>View IMDB's list of the most popular shows. This feature uses IMDB's MOVIEMeter algorithm to identify popular TV Shows.</p>
            </div>
        </app-link>

        <app-link href="addShows/popularAnime/" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addanime"></div></div>
            <div class="buttontext">
                <h3>Add From Anidb's Hot Anime list</h3>
                <p>View Anidb's list of the most popular anime shows. Anidb provides lists for Popular Anime, using the "Hot Anime" list.</p>
            </div>
        </app-link>
    </div>
</template>
<script>
import AppLink from './app-link.vue';

export default {
    name: 'addRecommended',
    components: {
        AppLink
    },
    computed: {
        header() {
            return this.$route.meta.header;
        }
    }
};
</script>
<style>
/* placeholder */
</style>
