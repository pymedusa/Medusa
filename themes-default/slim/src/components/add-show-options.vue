<template>
    <div id="add-show-options-content">
        <fieldset class="component-group-list">
            <div class="form-group">
                <div class="row">
                    <label for="customQuality" class="col-sm-2 control-label">
                        <span>Quality</span>
                    </label>
                    <div class="col-sm-10 content">
                        <quality-chooser :overall-quality="defaultConfig.quality" @update:quality:allowed="quality.allowed = $event" @update:quality:preferred="quality.preferred = $event"></quality-chooser>
                    </div>
                </div>
            </div>

            <div v-if="subtitlesEnabled" id="use-subtitles">
                <config-toggle-slider label="Subtitles" id="subtitles" :checked="defaultConfig.subtitles" @update="selectedSubtitleEnabled = $event"
                    :explanations="['Download subtitles for this show?']">
                </config-toggle-slider>
            </div>

           <div class="form-group">
                <div class="row">
                    <label for="defaultStatus" class="col-sm-2 control-label">
                        <span>Status for previously aired episodes</span>
                    </label>
                    <div class="col-sm-10 content">
                        <select id="defaultStatus" class="form-control form-control-inline input-sm" v-model="selectedStatus">
                            <option v-for="option in defaultEpisodeStatusOptions" :value="option.value" :key="option.value">{{ option.text }}</option>
                        </select>
                    </div>
                </div>
            </div>

            <div class="form-group">
                <div class="row">
                    <label for="defaultStatusAfter" class="col-sm-2 control-label">
                        <span>Status for all future episodes</span>
                    </label>
                    <div class="col-sm-10 content">
                        <select id="defaultStatusAfter" class="form-control form-control-inline input-sm" v-model="selectedStatusAfter">
                            <option v-for="option in defaultEpisodeStatusOptions" :value="option.value" :key="option.value">{{ option.text }}</option>
                        </select>
                    </div>
                </div>
            </div>

            <config-toggle-slider label="Season Folders" id="season_folders" :checked="defaultConfig.seasonFolders || namingForceFolders" :disabled="namingForceFolders"
                :explanations="['Group episodes by season folders?']" @update="selectedSeasonFoldersEnabled = $event">
            </config-toggle-slider>

            <config-toggle-slider v-if="enableAnimeOptions" label="Anime" id="anime" :checked="defaultConfig.anime"
                :explanations="['Is this show an Anime?']" @update="selectedAnimeEnabled = $event">
            </config-toggle-slider>

            <div v-if="enableAnimeOptions && selectedAnimeEnabled" class="form-group">
                <div class="row">
                    <label for="anidbReleaseGroup" class="col-sm-2 control-label">
                        <span>Release Groups</span>
                    </label>
                    <div class="col-sm-10 content">
                        <anidb-release-group-ui class="max-width" :blacklist="release.blacklist" :whitelist="release.whitelist"
                            :all-groups="release.allgroups" @change="onChangeReleaseGroupsAnime">
                        </anidb-release-group-ui>
                    </div>
                </div>
            </div>

            <config-toggle-slider label="Scene Numbering" id="scene" :checked="defaultConfig.scene"
                :explanations="['Is this show scene numbered?']" @update="selectedSceneEnabled = $event">
            </config-toggle-slider>

            <div class="form-group">
                <div class="row">
                    <label for="saveDefaultsButton" class="col-sm-2 control-label">
                        <span>Use current values as the defaults</span>
                    </label>
                    <div class="col-sm-10 content">
                        <button type="button" class="btn-medusa btn-inline" @click.prevent="saveDefaults" :disabled="saving || saveDefaultsDisabled">Save Defaults</button>
                    </div>
                </div>
            </div>
        </fieldset>
    </div>
</template>
<script>
import { mapState } from 'vuex';
import { apiRoute } from '../api';
import { combineQualities } from '../utils';
import ConfigToggleSlider from './config-toggle-slider.vue';
import AnidbReleaseGroupUi from './anidb-release-group-ui.vue';

export default {
    name: 'add-show-options',
    components: {
        AnidbReleaseGroupUi,
        ConfigToggleSlider
        // @TODO: Add `QualityChooser`
    },
    props: {
        showName: {
            type: String,
            default: '',
            required: false
        },
        enableAnimeOptions: {
            type: Boolean,
            default: false
        }
    },
    data() {
        return {
            saving: false,
            selectedStatus: null,
            selectedStatusAfter: null,
            quality: {
                allowed: [],
                preferred: []
            },
            selectedSubtitleEnabled: false,
            selectedSeasonFoldersEnabled: false,
            selectedAnimeEnabled: false,
            selectedSceneEnabled: false,
            release: {
                blacklist: [],
                whitelist: [],
                allgroups: []
            }
        };
    },
    mounted() {
        const { defaultConfig, update } = this;
        this.selectedStatus = defaultConfig.status;
        this.selectedStatusAfter = defaultConfig.statusAfter;
        this.$nextTick(() => update());

        this.$watch(vm => [
            vm.selectedStatus,
            vm.selectedStatusAfter,
            vm.selectedSubtitleEnabled,
            vm.selectedSeasonFoldersEnabled,
            vm.selectedSceneEnabled
        ].join(), () => {
            this.update();
        });
    },
    methods: {
        getReleaseGroups(showName) {
            const params = {
                series_name: showName // eslint-disable-line camelcase
            };

            return apiRoute
                .get('home/fetch_releasegroups', { params, timeout: 30000 })
                .then(response => response.data)
                .catch(error => {
                    this.$snotify.warning(
                        `Error while trying to fetch release groups for show "${showName}": ${error || 'Unknown'}`,
                        'Error'
                    );
                    console.warn(error);
                    return null;
                });
        },
        update() {
            const {
                selectedSubtitleEnabled,
                selectedStatus,
                selectedStatusAfter,
                selectedSeasonFoldersEnabled,
                selectedAnimeEnabled,
                selectedSceneEnabled,
                release,
                quality
            } = this;
            this.$nextTick(() => {
                this.$emit('change', {
                    subtitles: selectedSubtitleEnabled,
                    status: selectedStatus,
                    statusAfter: selectedStatusAfter,
                    seasonFolders: selectedSeasonFoldersEnabled,
                    anime: selectedAnimeEnabled,
                    scene: selectedSceneEnabled,
                    release,
                    quality
                });
            });
        },
        onChangeReleaseGroupsAnime(items) {
            this.release.whitelist = items.filter(item => item.memberOf === 'whitelist').map(item => item.name);
            this.release.blacklist = items.filter(item => item.memberOf === 'blacklist').map(item => item.name);
            this.update();
        },
        saveDefaults() {
            const {
                $store,
                selectedStatus,
                selectedStatusAfter,
                combinedQualities,
                selectedSubtitleEnabled,
                selectedSeasonFoldersEnabled,
                selectedAnimeEnabled,
                selectedSceneEnabled
            } = this;

            const section = 'main';
            const config = {
                showDefaults: {
                    status: selectedStatus,
                    statusAfter: selectedStatusAfter,
                    quality: combinedQualities,
                    subtitles: selectedSubtitleEnabled,
                    seasonFolders: selectedSeasonFoldersEnabled,
                    anime: selectedAnimeEnabled,
                    scene: selectedSceneEnabled
                }
            };

            this.saving = true;
            $store.dispatch('setConfig', { section, config }).then(() => {
                this.$snotify.success(
                    'Your "add show" defaults have been set to your current selections.',
                    'Saved Defaults'
                );
            }).catch(error => {
                this.$snotify.error(
                    'Error while trying to save "add show" defaults: ' + error.message || 'Unknown',
                    'Error'
                );
            }).finally(() => {
                this.saving = false;
            });
        }
    },
    computed: {
        ...mapState({
            defaultConfig: state => state.config.showDefaults,
            namingForceFolders: state => state.config.namingForceFolders,
            subtitlesEnabled: state => state.config.subtitles.enabled,
            episodeStatuses: state => state.statuses
        }),
        defaultEpisodeStatusOptions() {
            const { strings, values } = this.episodeStatuses;
            const { skipped, wanted, ignored } = values;
            return [skipped, wanted, ignored].map(value => ({
                value,
                text: strings[value]
            }));
        },
        /**
         * Calculate the combined value of the selected qualities.
         * @returns {number} - An unsigned integer.
         */
        combinedQualities() {
            const { quality } = this;
            const { allowed, preferred } = quality;
            return combineQualities(allowed, preferred);
        },
        /**
         * Check if any changes were made to determine if the "Save Defaults" button should be enabled.
         * @returns {boolean} - Should the save default buttons be disabled?
         */
        saveDefaultsDisabled() {
            const {
                enableAnimeOptions,
                defaultConfig,
                namingForceFolders,

                selectedStatus,
                selectedStatusAfter,
                combinedQualities,
                selectedSeasonFoldersEnabled,
                selectedSubtitleEnabled,
                selectedAnimeEnabled,
                selectedSceneEnabled
            } = this;

            return [
                selectedStatus === defaultConfig.status,
                selectedStatusAfter === defaultConfig.statusAfter,
                combinedQualities === defaultConfig.quality,
                selectedSeasonFoldersEnabled === (defaultConfig.seasonFolders || namingForceFolders),
                selectedSubtitleEnabled === defaultConfig.subtitles,
                !enableAnimeOptions || selectedAnimeEnabled === defaultConfig.anime,
                selectedSceneEnabled === defaultConfig.scene
            ].every(Boolean);
        }
    },
    asyncComputed: {
        releaseGroups() {
            const { selectedAnimeEnabled, showName } = this;
            if (!selectedAnimeEnabled || !showName) {
                return Promise.resolve([]);
            }

            return this.getReleaseGroups(showName).then(result => {
                if (result.groups) {
                    return result.groups;
                }
            });
        }
    },
    watch: {
        releaseGroups(groups) {
            this.release.allgroups = groups;
        },
        release: {
            handler() {
                this.update();
            },
            deep: true,
            immediate: false
        },
        /**
         * Whenever something changes that can impact the height of the component,
         * we need to update the parent formWizard, and make it resize.
         */
        quality: {
            handler() {
                this.$emit('refresh');
                this.update();
            },
            deep: true,
            immediate: false
        },
        selectedAnimeEnabled() {
            this.$emit('refresh');
            this.update();
        },
        defaultConfig(newValue) {
            this.selectedStatus = newValue.status;
            this.selectedStatusAfter = newValue.statusAfter;
        }
    }
};
</script>
<style>
/* placeholder */
</style>
