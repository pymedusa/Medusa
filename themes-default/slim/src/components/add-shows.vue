<template>
    <div id="addShowPortal">
        <app-link href="addShows/newShow/" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addnewshow"></div></div>
            <div class="buttontext">
                <h3>Add New Show</h3>
                <p>For shows that you haven't downloaded yet, this option finds a show on your preferred indexer, creates a directory for it's episodes, and adds it to Medusa.</p>
            </div>
        </app-link>

        <app-link href="addShows/existingShows/" id="btnExistingShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addexistingshow"></div></div>
            <div class="buttontext">
                <h3>Add Existing Shows</h3>
                <p>Use this option to add shows that already have a folder created on your hard drive. Medusa will scan your existing metadata/episodes and add the show accordingly.</p>
            </div>
        </app-link>
    </div>
</template>
<script>
import AppLink from './app-link.vue';

export default {
    name: 'addShows',
    components: {
        AppLink
    },
    computed: {
        header() {
            return this.$route.meta.header;
        }
    }
};
</script>
<style>
/* placeholder */
</style>
