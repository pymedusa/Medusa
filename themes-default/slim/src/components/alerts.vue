<template>
    <div v-if="system.branch !== null" id="alerts">
        <!-- BEGIN ALERTS -->
        <div v-if="system.branch !== 'master' && !developer" class="text-center">
            <div class="alert alert-danger upgrade-notification hidden-print" role="alert">
                <span>You're using the {{system.branch}} branch. Please use 'master' unless specifically asked</span>
            </div>
        </div>

        <div v-if="system.newestVersionMessage" class="text-center">
            <div class="alert alert-success upgrade-notification hidden-print" role="alert">
                <span v-html="system.newestVersionMessage" />
            </div>
        </div>
    </div>
</template>

<script>
import { mapState } from 'vuex';
export default {
    name: 'alerts',
    computed: {
        ...mapState({
            system: state => state.config.system,
            developer: state => state.config.general.developer
        })
    }
};
</script>

<style>
</style>
