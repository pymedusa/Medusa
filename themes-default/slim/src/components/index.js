export { default as AddRecommended } from './add-recommended.vue';
export { default as AddShowOptions } from './add-show-options.vue';
export { default as AddShows } from './add-shows.vue';
export { default as AnidbReleaseGroupUi } from './anidb-release-group-ui.vue';
export { default as AppFooter } from './app-footer.vue';
export { default as AppHeader } from './app-header.vue';
export { default as Backstretch } from './backstretch.vue';
export { default as Config } from './config.vue';
export { default as ConfigGeneral } from './config-general.vue';
export { default as ConfigPostProcessing } from './config-post-processing.vue';
export { default as ConfigNotifications } from './config-notifications.vue';
export { default as ConfigSearch } from './config-search.vue';
export { default as DisplayShow } from './display-show.vue';
export { default as EditShow } from './edit-show.vue';
export { default as History } from './history.vue';
export { default as Home } from './home.vue';
export { default as IRC } from './irc.vue';
export { default as Login } from './login.vue';
export { default as Logs } from './logs.vue';
export { default as ManualPostProcess } from './manual-post-process.vue';
export { default as RootDirs } from './root-dirs.vue';
export { default as Schedule } from './schedule.vue';
export { default as ShowHeader } from './show-header.vue';
export { default as ShowHistory } from './show-history.vue';
export { default as ShowResults } from './show-results.vue';
export { default as SnatchSelection } from './snatch-selection.vue';
export { default as Status } from './status.vue';
export { default as SubMenu } from './sub-menu.vue';
export { default as SubtitleSearch } from './subtitle-search.vue';
export * from './http';
export * from './helpers';
