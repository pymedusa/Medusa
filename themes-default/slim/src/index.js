import $ from 'jquery';
import 'bootstrap'; // eslint-disable-line import/no-unassigned-import
import 'bootstrap/dist/css/bootstrap.min.css'; // eslint-disable-line import/no-unassigned-import
import './css/open-sans.css'; // eslint-disable-line import/no-unassigned-import

import Vue from 'vue';
import Vuex from 'vuex';
import VueMeta from 'vue-meta';
import VueRouter from 'vue-router';
import VueNativeSock from 'vue-native-websocket';
import AsyncComputed from 'vue-async-computed';
import ToggleButton from 'vue-js-toggle-button';
import Snotify from 'vue-snotify';
import axios from 'axios';
import debounce from 'lodash/debounce';
import store from './store';
import router from './router';
import { isDevelopment } from './utils';
import { apiRoute, apiv1, api, webRoot, apiKey } from './api';
import { Asset, Backstretch, DisplayShow, PlotInfo, ShowSelector, ScrollButtons, FileBrowser, NamePattern, AppHeader, AppLink, SelectList, LanguageSelect, RootDirs, AnidbReleaseGroupUi, SnatchSelection } from './components';

if (window) {
    window.isDevelopment = isDevelopment;

    // Adding libs to window so mako files can use them
    window.$ = $;
    window.jQuery = $;
    window.Vue = Vue;
    window.Vuex = Vuex;
    window.VueMeta = VueMeta;
    window.VueRouter = VueRouter;
    window.VueNativeSock = VueNativeSock;
    window.AsyncComputed = AsyncComputed;
    window.ToggleButton = ToggleButton;
    window.Snotify = Snotify;
    window.axios = axios;
    window._ = { debounce };
    window.store = store;
    window.router = router;
    window.apiRoute = apiRoute;
    window.apiv1 = apiv1;
    window.api = api;

    window.MEDUSA = {
        common: {},
        config: {},
        home: {},
        manage: {},
        history: {},
        errorlogs: {},
        schedule: {},
        addShows: {}
    };
    window.webRoot = webRoot;
    window.apiKey = apiKey;
    window.apiRoot = webRoot + '/api/v2/';

    // Push pages that load via a vue file but still use `el` for mounting
    window.components = [];
    window.components.push(AnidbReleaseGroupUi);
    window.components.push(AppHeader);
    window.components.push(AppLink);
    window.components.push(Asset);
    window.components.push(Backstretch);
    window.components.push(DisplayShow);
    window.components.push(FileBrowser);
    window.components.push(LanguageSelect);
    window.components.push(NamePattern);
    window.components.push(PlotInfo);
    window.components.push(RootDirs);
    window.components.push(ScrollButtons);
    window.components.push(SelectList);
    window.components.push(ShowSelector);
    window.components.push(SnatchSelection);
}
const UTIL = {
    exec(controller, action) {
        const ns = MEDUSA;
        action = (action === undefined) ? 'init' : action;

        if (controller !== '' && ns[controller] && typeof ns[controller][action] === 'function') {
            ns[controller][action]();
        }
    },
    init() {
        if (typeof startVue === 'function') { // eslint-disable-line no-undef
            startVue(); // eslint-disable-line no-undef
        } else {
            $('[v-cloak]').removeAttr('v-cloak');
        }

        const { body } = document;
        $('[asset]').each(function() {
            const asset = $(this).attr('asset');
            const series = $(this).attr('series');
            const path = apiRoot + 'series/' + series + '/asset/' + asset + '?api_key=' + apiKey;
            if (this.tagName.toLowerCase() === 'img') {
                const defaultPath = $(this).attr('src');
                if ($(this).attr('lazy') === 'on') {
                    $(this).attr('data-original', path);
                } else {
                    $(this).attr('src', path);
                }
                $(this).attr('onerror', 'this.src = "' + defaultPath + '"; return false;');
            }
            if (this.tagName.toLowerCase() === 'a') {
                $(this).attr('href', path);
            }
        });
        const controller = body.getAttribute('data-controller');
        const action = body.getAttribute('data-action');

        UTIL.exec('common'); // Load common
        UTIL.exec(controller); // Load MEDUSA[controller]
        UTIL.exec(controller, action); // Load MEDUSA[controller][action]

        window.dispatchEvent(new Event('medusa-loaded'));
    }
};

$.fn.extend({
    addRemoveWarningClass(_) {
        if (_) {
            return $(this).removeClass('warning');
        }
        return $(this).addClass('warning');
    }
});

const { pathname } = window.location;
if (!pathname.includes('/login') && !pathname.includes('/apibuilder')) {
    api.get('config/main').then(response => {
        $.extend(MEDUSA.config, response.data);
        MEDUSA.config.themeSpinner = MEDUSA.config.themeName === 'dark' ? '-dark' : '';
        MEDUSA.config.loading = '<img src="images/loading16' + MEDUSA.config.themeSpinner + '.gif" height="16" width="16" />';

        if (navigator.userAgent.indexOf('PhantomJS') === -1) {
            $(document).ready(UTIL.init);
        }

        MEDUSA.config.indexers.indexerIdToName = indexerId => {
            if (!indexerId) {
                return '';
            }
            return Object.keys(MEDUSA.config.indexers.config.indexers).filter(indexer => { // eslint-disable-line array-callback-return
                if (MEDUSA.config.indexers.config.indexers[indexer].id === parseInt(indexerId, 10)) {
                    return MEDUSA.config.indexers.config.indexers[indexer].name;
                }
            })[0];
        };

        MEDUSA.config.indexers.nameToIndexerId = name => {
            if (!name) {
                return '';
            }
            return MEDUSA.config.indexers.config.indexers[name];
        };
    }).catch(error => {
        console.debug(error);
        alert('Unable to connect to Medusa!'); // eslint-disable-line no-alert
    });
}
