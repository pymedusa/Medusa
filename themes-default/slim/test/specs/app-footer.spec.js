import Vuex, { Store } from 'vuex';
import VueRouter from 'vue-router';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import tk from 'timekeeper';

import { AppFooter } from '../../src/components';
import constsModule from '../../src/store/modules/consts';
import systemModule from '../../src/store/modules/system';
import fixtures from '../__fixtures__/common';

describe('AppFooter', () => {
    let localVue;
    let store;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuex);
        localVue.use(VueRouter);

        const { state } = fixtures;
        store = new Store({
            modules: {
                consts: constsModule,
                system: systemModule
            }
        });
        store.replaceState(state);
    });

    it('renders', () => {
        tk.travel(new Date(2019, 6, 10, 9, 51, 21, 300));

        const wrapper = shallowMount(AppFooter, {
            localVue,
            store
        });

        expect(wrapper.element).toMatchSnapshot();

        tk.reset();
    });
});
