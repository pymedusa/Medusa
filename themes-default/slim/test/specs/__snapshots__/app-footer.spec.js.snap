// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter renders 1`] = `
<footer>
  <div
    class="footer clearfix"
  >
    <span
      class="footerhighlight"
    >
      2
    </span>
     Shows (
    <span
      class="footerhighlight"
    >
      2
    </span>
     Active)
        | 
    <span
      class="footerhighlight"
    >
      100
    </span>
     
    <span
      class="footerhighlight"
    >
      <app-link-stub
        href="manage/episodeStatuses?whichStatus=2"
        placeholder="indexer-to-name"
        title="View overview of snatched episodes"
      >
        +5
      </app-link-stub>
    </span>
    
            Snatched
        
        / 
    <span
      class="footerhighlight"
    >
      150
    </span>
     Episodes Downloaded 
    <span
      class="footerhighlight"
    >
      (66.7%)
    </span>
    
        | Daily Search: 
    <span
      class="footerhighlight"
    >
      0:39:46
    </span>
    
        | Backlog Search: 
    <span
      class="footerhighlight"
    >
      11:59:46
    </span>
     
    <div>
      
                Memory used: 
      <span
        class="footerhighlight"
      >
        56.85 MB
      </span>
       |
             
            Branch: 
      <span
        class="footerhighlight"
      >
        master
      </span>
       |
            Now: 
      <span
        class="footerhighlight"
      >
        Wednesday, 07/10/19 09:51:21
      </span>
    </div>
  </div>
</footer>
`;
